### DISCUSSION TITLE
z2mqtt not finding Sonoff USB (with solution)

### DISCUSSION THREAD

**Discussion by kerryland** [Koenkk/zigbee2mqtt#17846] [Devices]
Today I decided it was time to do some house-keeping and bump up versions of my OS [dietpi](https://dietpi.com/) and Home Assistant, the end result of which was that zigbee2mqtt wouldn't start up.

```
Zigbee2MQTT:error 2023-05-31 19:33:46: Error: Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_be08a0cbe021ec1182bf20c32c86906c-if00-port0'
```

Weird, because logs looked ok:

```
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: new full-speed USB device number 9 using xhci_hcd
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: New USB device found, idVendor=10c4, idProduct=ea60, bcdDevice= 1.00
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: Product: Sonoff Zigbee 3.0 USB Dongle Plus
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: Manufacturer: ITead
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: SerialNumber: be08a0cbe021ec1182bf20c32c86906c
May 31 19:17:05 blackbox kernel: usb 1-1.2.4: cp210x converter now attached to ttyUSB0
```
On closer inspection I discovered that the error was true. `/dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_be08a0cbe021ec1182bf20c32c86906c-if00-port0` really did not exist anymore.

The fix was simple enough: point Zigbee2mqtt at the "new" device that appeared in `/dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.2.4:1.0-port0`, aka `/dev/ttyUSB0`.

...which I did by visiting the Zigbee2Mqtt "Addon" page in Home Assistant , clicking "Configuration",  and changing `port: /dev/ttyUSB0`

Interestingly all my "Router" devices needed to be power-cycled before they would reconnect, but from then on it was business as usual.

Hopefully this helps someone else with the same problem.

**Comment by Bmollous** [comment]
Solved the issue for me thanks
